.github/workflows/pre-commit.yml

```
@@ -1,42 +1,75 @@
 name: Lint Check (pre-commit)
 
 on:
-  pull_request:
+  # We have to use pull_request_target here as pull_request does not grand
+  # enough permission for reviewdog
+  pull_request_target:
   push:
 
 jobs:
-  pre-commit:
+  pre-commit-push:
+    name: Pre-Commit check on Push
     runs-on: ubuntu-latest
-    name: pre-commit
+    if: ${{ github.event_name == 'push' }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.13
+
+        # We wish to run pre-commit on all files instead of the changes
+        # only made in the push commit.
+        #
+        # So linting error persists when there's formatting problem.
+      - uses: pre-commit/action@v3.0.1
+
+  pre-commit-pr:
+    name: Pre-Commit check on PR
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'pull_request_target' }}
+
     permissions:
       contents: read
       checks: write
       issues: write
       pull-requests: write
 
     steps:
-      - name: Checkout code
+      - name: Checkout repository
         uses: actions/checkout@v4
 
+        # pull_request_target checkout the base of the repo
+        # we need to checkout the actual pr to lint the changes
```

```suggestion
        # We need to checkout the actual pr to lint the changes.
```
Please update comments to be uniform.

---

Ah, nice catch, thanks.